src/libponyc/codegen/genexe.cc

```diff
@@ -281,6 +286,106 @@ static const char* env_cc_or_pony_compiler(bool* out_fallback_linker)
 }
 #endif
 
+/*
+ * search_path (called from search_paths)
+ *  - A single tuple (path, depth) to search
+ *  - The target filename to find
+ *  - Whether to include the filename or not
+ *
+ * There are two types of searches that we do.  One to find the paths
+ * so we can add them via -L (libpthread, libm, librt, libatomic etc),
+ * the other for identifying the paths to specific object files such
+ * as crtn.o et al).
+ *
+ * I think this should be refactored into two separate functions as
+ * we should probably deduplicate the -L paths before push_backing them
+ * onto args.
+ */
+std::optional<std::string> search_path(std::tuple<fs::path, int> search_path,
+                                       std::string targetstring,
+                                       bool include_filename)
+{
+  std::optional<std::string> result = std::nullopt;
+
+  fs::directory_options options =
+    (fs::directory_options::follow_directory_symlink |
+     fs::directory_options::skip_permission_denied);
+
+  /*
+   * Even though we instruct recursive_directory_iterator to just skip directories
+   * that we don't have permission to enter, if we don't have permission for the
+   * initially specified file path, or the filepath doesn't exist - then it will
+   * raise an exception.
+   *
+   * This is expected as we'll almost certainly search paths that don't exist on
+   * other distributions.
+   */
+  std::error_code ec;
+  for (auto iter = fs::recursive_directory_iterator(std::get<0>(search_path), options, ec);
+       iter != fs::recursive_directory_iterator(); ++iter) {
+    if (iter.depth() == std::get<1>(search_path)) {
+      iter.disable_recursion_pending();
+    };
```

Comment:
1. our standard style is no {} for single line ifs.
2. we don't do ; after the }